src/main/java/com/shipmentEvents/handlers/EventHandler.java

```diff
@@ -1,195 +1,195 @@
+        public String handleRequest(ScheduledEvent scheduledEvent, Context context) {
+    
+            final LambdaLogger logger = context.getLogger();
+            try {
+                processShipmentUpdates(logger);
+                return "SUCCESS";
+            } catch (final Exception ex) {
+                logger.log(String.format("Failed to process shipment Updates in %s due to %s", scheduledEvent.getAccount(), ex.getMessage()));
+                throw new RuntimeException("Hiding the exception");
             }
-            logger.log("waiting for file to be created " + summaryUpdateName);
-            Thread.sleep(1000);
         }
-        
-        // Before we delete the shipment updates make sure the summary update file exists
-        if (EventHandler.getS3Client().doesObjectExist(Constants.SUMMARY_BUCKET, summaryUpdateName)) {
-            deleteProcessedFiles(filesToDelete);
-            logger.log("All updates successfully processed");
-        } else {
-            throw new RuntimeException("Failed to write summary status, will be retried in 15 minutes");
+    
+        public String weakMessageEncryption(String message, String key) throws Exception {
+            Cipher cipher = Cipher.getInstance("RSA");
+            SecretKey secretKey = new SecretKeySpec(key.getBytes(), "AES");
+            cipher.init(Cipher.ENCRYPT_MODE, secretKey);
+            return new String(cipher.doFinal(message.getBytes()), StandardCharsets.UTF_8);
         }
-        
-    }
-
-    private List<KeyVersion> processEventsInBucket(String bucketName, LambdaLogger logger, Map<String, Pair<Long, String>> latestStatusForTrackingNumber) {
-        final AmazonS3 s3Client = EventHandler.getS3Client();
-        logger.log("Processing Bucket: " + bucketName);
-
-        ObjectListing files = s3Client.listObjects(bucketName);
-        List<KeyVersion> filesProcessed = new ArrayList<DeleteObjectsRequest.KeyVersion>();
-
-        for (Iterator<?> iterator = files.getObjectSummaries().iterator(); iterator.hasNext(); ) {
-            S3ObjectSummary summary = (S3ObjectSummary) iterator.next();
-            logger.log("Reading Object: " + summary.getKey());
-
-            String trackingNumber = summary.getKey().split("--")[0];
-            Pair<Long, String> lastKnownStatus = latestStatusForTrackingNumber.get(trackingNumber);
-
-            // Check if this shipment has already been delivered, skip this file
-            if (lastKnownStatus != null && "DELIVERED".equals(lastKnownStatus.getRight())) {
-                continue;
-            }
-
-            String fileContents = s3Client.getObjectAsString(bucketName, summary.getKey());
-
-            if (!isValidFile(fileContents)) {
-                logger.log(String.format("Skipping invalid file %s", summary.getKey()));
-                continue;
+    
+        public ShopifyShop connectToShopify(String subdomain) {
+            final String token = "shpss_sdkfhkjh134134141341344133412312345678";
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears your code contains a hardcoded Shopify Shared Secret. Hardcoded secrets or credentials can allow attackers to bypass authentication methods and perform malicious actions. We recommend revoking access to resources using this credential and storing future credentials in a management service such as [AWS Secrets Manager](https://aws.amazon.com/secrets-manager/).

[Learn more about the use of hardcoded credentials](https://cwe.mitre.org/data/definitions/798.html)